### PR TITLE
#6Task. Alexei.Kolgan

### PR DESCRIPTION
**Pull request by kolganem**
- https://travis-ci.com/github/kolganem/js-assignments/builds/174210608

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -46,7 +56,10 @@ function getFizzBuzz(num) {
  *   10 => 3628800
  */
 function getFactorial(n) {
-    throw new Error('Not implemented');
+    let rval=1;
+    for (let i = 2; i <= n; i++)
+        rval = rval * i;
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can write it in this way also :)
rval *= i;

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -63,7 +76,11 @@ function getFactorial(n) {
  *   -1,1  =>  0  ( = -1 + 0 + 1 )
  */
 function getSumBetweenNumbers(n1, n2) {
-    throw new Error('Not implemented');
+    let res = 0;
+    for(let i = n1; i <= n2; i++)
+        res = res + i;
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can write it in this way also :)
res += i;

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -81,8 +98,13 @@ function getSumBetweenNumbers(n1, n2) {
...
  */
-function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+function isTriangle(a,b,c) 
+{
+    return ((Math.pow(a, 2) + Math.pow(b, 2) === Math.pow(c, 2)) ||
+        (Math.pow(b, 2) + Math.pow(c, 2) === Math.pow(a, 2)) ||
+        (Math.pow(c, 2) + Math.pow(a, 2) === Math.pow(b, 2)) ||
+        (a === b && a === c));
+
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js:103]: Cool, but you can do it in 1 row. Please, try to improve your code. Thank you!

Comment (kolganem) [reply task/06-conditions-n-loops-tasks.js:103]: I thought is the string is big it is better to split it.

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -209,7 +256,11 @@ function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
  * 'noon' => 'noon'
  */
 function reverseString(str) {
-    throw new Error('Not implemented');
+
+    let strForSplit = str.split("");   
+    let arrForReverse = strForSplit.reverse();    
+    let resultStr = arrForReverse.join(""); 
+    return resultStr;
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can write it all in 1 row :)

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -226,7 +277,13 @@ function reverseString(str) {
...
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+
+    let strForSplit = String(num).split("");   
+    let arrForReverse = strForSplit.reverse();    
+    let resultStr = arrForReverse.join("");
+
+    return Number(resultStr);
+
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: Same. You can write it all in 1 row

File: task/06-conditions-n-loops-tasks.js (hunk 3 above)
Comment (elizavetaShcherbatykhAis) [reply task/06-conditions-n-loops-tasks.js:103]: No, I mean you can do it in another way and it will be much shorter

Comment (elizavetaShcherbatykhAis) [reply task/06-conditions-n-loops-tasks.js:103]: You don't need Math.pow() here :)

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -270,7 +343,16 @@ function isCreditCardNumber(ccn) {
...
 function getDigitalRoot(num) {
-    throw new Error('Not implemented');
+    let resultSum = num;
+    let newArr = [];
+    let reducer = (a,b) => parseInt(a) + parseInt(b);
+
+    while (resultSum > 9) 
+    {
+        newArr = resultSum.toString().split("")
+        resultSum = newArr.reduce(reducer)
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can write it like:
resultSum = resultSum.toString().split("").reduce(reducer)
So newArr param is no needed anymore and your code is shorter :)

**Comment by elizavetaShcherbatykhAis** [comment]
Well done! Please check the comments
I'll mark the PR as draft please click "ready for review" when it will be corrected!
Thank you!

File: task/06-conditions-n-loops-tasks.js (hunk 3 above)
Comment (kolganem) [reply task/06-conditions-n-loops-tasks.js:103]: done

File: task/06-conditions-n-loops-tasks.js (hunk 4 above)
Comment (kolganem) [reply task/06-conditions-n-loops-tasks.js]: done

**Comment by kolganem** [comment]
Thanks for review :)

File: task/06-conditions-n-loops-tasks.js (hunk 5 above)
Comment (kolganem) [reply task/06-conditions-n-loops-tasks.js]: done

File: task/06-conditions-n-loops-tasks.js (hunk 6 above)
Comment (kolganem) [reply task/06-conditions-n-loops-tasks.js]: done

**Comment by kolganem** [comment]
Sorry, fixed some misprint from Travic CI. Now everything is all right.
https://travis-ci.com/github/kolganem/js-assignments/builds/176311564

**Comment by elizavetaShcherbatykhAis** [comment]
Super! I close PR as accepted

**Pull request closed without merging**